eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -45,25 +45,25 @@ parameters:
           - eng/common/TestResources/sub-config/AzurePublicMsft.json
       Preview:
         SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources-preview)
```

Comment:
Can we remove the sub config as well or is there still stuff in there that isn't in the commited config?

Reply:
I was thinking I can either be additive, since go hasn't gone to fed auth by default yet, or I can make all the changes here dependent on fed auth (which I think is required for looking at the git config path behavior?)

Reply:
Would it make things easier to go to default on? @chlowell can comment if we are ready for that.

Reply:
@benbp by having both configs you cannot work with both modes because the file sub config clears the client secret. So we will have to be in one world or the other.

Reply:
> Would it make things easier to go to default on? @chlowell can comment if we are ready for that.

@RickWinter yes a bit simpler, I'm happy to roll those UseFederatedAuth changes into this PR set as well if you're ready.

Reply:
I believe we're ready for that. All our live tests should just work when switched to federated auth because they use the test credential factory returning `AzurePipelinesCredential` when running in an AzurePowerShell@5 task.